code/game/objects/items/weapons/misc.dm

```diff
@@ -106,27 +106,22 @@
 	playsound(loc, 'sound/weapons/energy_blast.ogg', 50, TRUE)
 	playsound(loc, 'sound/weapons/genhit2.ogg', 50, TRUE)
 	var/atom/throw_target = get_edge_target_turf(M, get_dir(src, get_step_away(M, src)))
-	M.throw_at(throw_target, 5 * setting, 0.5 + (setting / 2))
+	var/throw_distance = setting * LERP(5 , 2, M.mob_size / MOB_SIZE_BIG) 
+	M.throw_at(throw_target, throw_distance, 0.5 + (setting / 2))
 	cell.charge -= powerused
 	return ..()
 
 /obj/item/weapon/powerfist/attackby(obj/item/I, mob/user, params)
-	if(!istype(I, /obj/item/cell))
+	if(!istype(I, /obj/item/cell/lasgun))
 		return ..()
 	if(cell)
-		to_chat(user, "<span class='warning'>There already is a cell installed!</span>")
-		return
+		unload(user)
 	user.transferItemToLoc(I,src)
 	cell = I
 	to_chat(user, "<span class='notice'>You insert the [I] into the [src].</span>")
 
-/obj/item/weapon/powerfist/screwdriver_act(mob/living/user, obj/item/I)
-	if(..())
-		return TRUE
-	if(!cell)
-		to_chat(user, "<span class='notice'>There is no cell installed!</span>")
-		return TRUE
+/// Remove the cell from the powerfist
+/obj/item/weapon/powerfist/proc/unload(mob/user)
```

Comment:
Is the only way to get a cell out to replace it?

Reply:
yeah. I can put back the screwdriver thing, i forgot that you can recharge the battery